docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
+contributor.
+
+We currently build images in CI and run tests as part of that, we should only run tests as a primary
+task.
+
+Downside of current setup:
+
+- CircleCI requires delegation of access to GitHub
+  - for releases we use a token
+  - we don't have a way to automate workflows that require pushing to the repo
+- CircleCI cache 
+  - it is not very fast, it turns out pulling image with cached dependencies is as fast
```

Comment:
what does not being very fast mean? compared to what? I am not sure i understand this sentence as a whole.

Reply:
Right now we do this:

- restore cache from previous build
- mount cache as docker volume
- run tests and build new binary

In my experiment it's as fast (or faster) to do the following:

- pull docker image
- run tests and build new binary

So, `docker pull` for a large image that contains all cached objects takes as long (or less time) then using CircleCI cache feature. There are other benefits to this, as we can tell exactly what's being cached and can reproduce it anywhere else.

Reply:
Please, add this explanation to the doc, sounds good.

Reply:
Bonus points for:

- mentioning build time before this change,
- mentioning time after this change,
- mentioning this in `docs/proposal-006-build-improvements.md`.